webapp/services.py

```diff
@@ -11,19 +11,24 @@
 # Local
 from webapp.database import db_session
 from webapp.lib.processors import ImageProcessor
-from webapp.models import Asset
+from webapp.models import Asset, Tag
 from webapp.swift import file_manager
 
 
 class AssetService:
-    def find_assets(self, file_type="%", query=None):
+    def find_assets(self, file_type="%", tag=None, query=None):
         """
         Find assets that matches the given criterions
         """
-        if not query:
-            return db_session.query(Asset).all()
-        if file_type == "":
+        if not query or query == "":
```

Comment:
`""` is falsy in python so I think `not query` is enough

Reply:
Good to know, thanks!